### PR TITLE
Migrate from TestNG to Junit 5

### PR DESCRIPTION
**Pull request by Stephan202**

### CODE REVIEW

File: rxjava-support/src/test/java/tech/picnic/rx/PicnicRxPluginsTest.java [hunk 1]
```diff
@@ -50,14 +50,15 @@ public void testPropagate() {
     verifyVerificationCounter(ctx2, 3);
 
     // With context propagation
-    init();
+    setUp();
```
Comment (Stephan202) [inline rxjava-support/src/test/java/tech/picnic/rx/PicnicRxPluginsTest.java]: It's a bit wonky that we call this method in the middle of a test; should be refactored in the context of another PR.

Comment (philleonard) [reply rxjava-support/src/test/java/tech/picnic/rx/PicnicRxPluginsTest.java]: 👍 Is the rename following Junit standards?

File: rxjava-support/src/test/java/tech/picnic/rx/PicnicRxPluginsTest.java [hunk 2]
```diff
@@ -50,14 +50,15 @@ public void testPropagate() {
...
     obs.subscribeOn(Schedulers.io())
         .doOnNext(i -> verifyActive(ctx1))
         .ignoreElements()
         .blockingAwait();
     verifyVerificationCounter(ctx1, 3);
   }
 
+  @Test
```
Comment (philleonard) [inline rxjava-support/src/test/java/tech/picnic/rx/PicnicRxPluginsTest.java]: Shame we lose class level annotations for `Test`

File: rxjava-support/src/test/java/tech/picnic/rx/PicnicRxPluginsTest.java (hunk 1 above)
Comment (Stephan202) [reply rxjava-support/src/test/java/tech/picnic/rx/PicnicRxPluginsTest.java]: Yep, it's what one of the PMD rules enabled by default in Codility suggested. (Now it still complains about a bunch of other violations we don't care about; will review that before merging this PR.)

File: rxjava-support/src/test/java/tech/picnic/rx/PicnicRxPluginsTest.java (hunk 2 above)
Comment (Stephan202) [reply rxjava-support/src/test/java/tech/picnic/rx/PicnicRxPluginsTest.java]: Yep.

File: pom.xml [hunk 3]
```diff
@@ -174,6 +175,11 @@
                 <artifactId>java-hamcrest</artifactId>
                 <version>2.0.0.0</version>
             </dependency>
+            <dependency>
+                <groupId>org.junit.jupiter</groupId>
+                <artifactId>junit-jupiter-api</artifactId>
+                <version>${version.junit-jupiter}</version>
+            </dependency>
```
Comment (nathankooij) [inline pom.xml]: Maybe consider using the JUnit BOM (https://junit.org/junit5/docs/current/user-guide/#dependency-metadata-junit-bom).

Comment (Stephan202) [reply pom.xml]: Nice one; added a commit. Couldn't nicely get rid of `${version.junit-jupiter}`, because the version is still shared with the Surefire dependency declaration.

**Comment by Stephan202** [comment]
PR is now on top of #31 in order to simplify the setup. Some tests (still? I don't remember) fail; will get back to this.

File: rxjava-support/src/test/java/tech/picnic/rx/PicnicRxPluginsTest.java [hunk 4]
```diff
@@ -17,26 +17,26 @@
...
+  static void tearDown() {
     PicnicRxPlugins.unsetContextPropagation();
   }
 
-  public void testPropagate() {
+  @Test
+  void testPropagate() {
     Observable<Integer> obs = Observable.just(1, 2, 3);
 
     // Without context propagation
```
[the remark this answers is not in the document]
Comment (Stephan202) [reply rxjava-support/src/test/java/tech/picnic/rx/PicnicRxPluginsTest.java]: `RxJavaPlugins.setScheduleHandler(null)` is indirectly called by `PicnicRxPlugins.unsetContextPropagation()`, so calling `tearDown()` is indeed nicer for symmetry.

**Comment by Stephan202** [comment]
The Codacy failure is because it doesn't like Java's implicit package-private scope; that doesn't make any sense. Will just merge and separately think about how to better configure `Codacy` (or drop it altogether).
